src/components/FormAlertWithSubmitButton.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import {StyleProp, View, ViewStyle} from 'react-native';
+import useThemeStyles from '@styles/useThemeStyles';
+import Button from './Button';
+import FormAlertWrapper from './FormAlertWrapper';
+
+type FormAlertWithSubmitButtonProps = {
+    /** Error message to display above button */
+    message?: string;
+
+    /** Whether the button is disabled */
+    isDisabled?: boolean;
+
+    /** Whether message is in html format */
+    isMessageHtml?: boolean;
+
+    /** Styles for container element */
+    containerStyles?: StyleProp<ViewStyle>;
+
+    /** Is the button in a loading state */
+    isLoading?: boolean;
+
+    /** Callback fired when the "fix the errors" link is pressed */
+    onFixTheErrorsLinkPressed?: () => void;
+
+    /** Submit function */
+    onSubmit: () => void;
+
+    /** Should the button be enabled when offline */
+    enabledWhenOffline?: boolean;
+
+    /** Disable press on enter for submit button */
+    disablePressOnEnter?: boolean;
+
+    /** Whether the form submit action is dangerous */
+    isSubmitActionDangerous?: boolean;
+
+    /** Custom content to display in the footer after submit button */
+    footerContent?: React.ReactNode;
+
+    /** Styles for the button */
+    buttonStyles?: StyleProp<ViewStyle>;
+
+    /** Whether to show the alert text */
+    isAlertVisible: boolean;
+
+    /** Text for the button */
+    buttonText: string;
+};
+
+function FormAlertWithSubmitButton({
+    message = '',
+    isDisabled = false,
+    isMessageHtml = false,
+    containerStyles,
```

Comment:
Careful, this prop used to default to array

Reply:
@pac-guerreiro we agreed earlier that styles props don't need default values [here](https://callstack-hq.slack.com/archives/C05P72RDZD2/p1700214784915519) is a conversation

Reply:
@kubabutkiewicz Oh okay, maybe that discussion should be added to the TS guidelines, what you think @fabioh8010 ?